### PR TITLE
[DOCUMENTED AND READY TO MINGLE](EPIC MEME VIDEO INSIDE!) NEW COOL ADMIN TOOL: ADVANCED HIGH DEFINITION SUPPLYPOD SUMMONING (also supplypod refactor) (PLEASE UPTHUMB)

### PR DESCRIPTION
**Pull request by MrDoomBringer**
### **[WATCH THIS EPIC MEME SHOWCASE VIDEO AND DONT FORGET TO LIKE SUBSCRIBE AND UPTHUMB THIS PR](https://youtu.be/UijhSBDELU8)**

there are too many cool features to go over here so just check out the enclosed instruction video (where i only show off like 3 features but whatever) ^

image

:cl: MrDoomBringer
admin: Admins can now spawn things in ICly (as well as do a bunch of other cool new stuff) using the Config/Launch Supplypod verb!

code: also supplypods have been refactored 
/:cl:

this has been done for ages i finally got around to actually PRing it. theres still like 1 or 2 minor things i gotta fix before its merged tho. 

---

**TODO:**
- [x] add comments/documentation
- [x] add a way to target mobs (to fully replace the supplypod smite)
- [x] fix holo-med sprite deletion
- [x] make supplypods come in at an angle
- [x] fix descs of custom supplypods
- [x] if 40105 gets merged first, add a message to the buildmode var that this exists
- [x] add a cult pod
- [x] and a missle pod
- [x] add a "do not open" option
- [x] add a custom name button
- [x] add a cool centcom pod holder facility
- [x] fix missile door
---

@admins waddya think?
if u send suggestions ill literally add them

### CODE REVIEW

**Comment by Therapist1** [comment]
best meme ive ever seen, upthumbed the pr, subscribed the pr, commented the pr.

**Comment by MrDoomBringer** [comment]
@therapist1 that's the kinda attitude this server needs

File: code/modules/cargo/gondolapod.dm
Comment (81Denton) [inline code/modules/cargo/gondolapod.dm]: This one's missing `/obj/item/reagent_containers/food/snacks/meat/slab/gondola = 1`

**Comment by 81Denton** [comment]
Very good PR makes me wish I was an admin so I could launch this at Helen Weinstein

**Comment by 81Denton** [comment]
I thunk you accidentally removed `holo_medical` and `holo_medical-deny` icons

**Comment by davethwave** [comment]
that gondola pod though...

**Comment by deathride58** [comment]
You spent so much time thinking of if you could, but never stopped for just one moment to think if you should

**Comment by MrDoomBringer** [comment]
@Denton I think ur right

**Comment by Arianya** [comment]
Glad you found the time to do this, thanks for your time and effort 👍

**Comment by Nabski89** [comment]
If you set it to explode on landing, will it set off explosives inside the pod?

How many pod options do we have exactly? I'm not sure what the advanced style does and remembering you having a BUNCH of sprites.

File: code/modules/cargo/expressconsole.dm
Comment (octareenroon91) [inline code/modules/cargo/expressconsole.dm]: Please don't merge with the old comment in place.

**Comment by octareenroon91** [comment]
>If you set it to explode on landing, will it set off explosives inside the pod?

Pod has do-nothing ex_act() and contents_explosion()

File: code/modules/admin/verbs/randomverbs.dm
Comment (octareenroon91) [inline code/modules/admin/verbs/randomverbs.dm]: The `2` is for pod bay 2 ?

File: code/modules/cargo/expressconsole.dm
Comment (MrDoomBringer) [reply code/modules/cargo/expressconsole.dm]: oh yeah i havent done anything with comments yet

File: code/modules/admin/verbs/randomverbs.dm
Comment (MrDoomBringer) [reply code/modules/admin/verbs/randomverbs.dm]: whups forgot to change that back

**Comment by MrDoomBringer** [comment]
>How many pod options do we have exactly? I'm not sure what the advanced style does and remembering you having a BUNCH of sprites.

You have as many as there are visible on that screenshot. Ill add more if you suggest em, only takes a few seconds.

Also, each button listed comes with a tooltip explaining exactly what it does, so no worries about forgetting what something is.

File: code/modules/tgui/tgui.dm
Comment (optimumtact) [inline code/modules/tgui/tgui.dm]: the hell is this

Comment (Supermichael777) [reply code/modules/tgui/tgui.dm]: it closes src_object' associated ui, duh

come to think if it i don't think ui objects ever GC

**Comment by lolman360** [comment]
move aside oranges, we have a new best coder in the house

**Comment by TheDreamweaver** [comment]
Merge ready?

**Comment by MrDoomBringer** [comment]
@TheDreamweaver nope but soon

**Comment by DaxDupont** [comment]
Make a Nyan catgirl pod you absolute mad man.

Also a cult one

**Comment by MrDoomBringer** [comment]
@DaxDupont 
i take no pleasure in this
image

**Comment by 81Denton** [comment]
the pod has no mouth and it must scream

**Comment by MrDoomBringer** [comment]
@81Denton 
do i do it like this y/n
image

**Comment by MrDoomBringer** [comment]
cult one comin soon(tm)

**Comment by 81Denton** [comment]
AAAHHH

**Comment by 81Denton** [comment]
I'd rather have seen a goatse pod

**Comment by optimumtact** [comment]
@MrDoomBringer remove the catgirl pod sprite

**Comment by octareenroon91** [comment]
harsh

**Comment by MrDoomBringer** [comment]
based fucking orange lmao

**Comment by MrDoomBringer** [comment]
@optimumtact that thing has been removed thank god
a new supplypod has arrived to fill the void
image

**Comment by TheDreamweaver** [comment]
@mrdoombringer I love this so much. I must see the open state.

**Comment by MrDoomBringer** [comment]
@TheDreamweaver 
image
AAAA

**Comment by TheDreamweaver** [comment]
Omg, marry me

**Comment by MrDoomBringer** [comment]
no u

**Comment by TheDreamweaver** [comment]
Let's be honest, you already had me at the gondola pod

**Comment by octareenroon91** [comment]
Was that a unicorn on the No-No-Neko pod?

**Comment by MrDoomBringer** [comment]
@octareenroon91 it was meant to be a bow but now i def see it as a unicorn horn
welp good thing its gone now

**Comment by optimumtact** [comment]
can you make your PR compile and wait for review

**Review by octareenroon91 — CHANGES_REQUESTED** [review]

**Comment by MrDoomBringer** [comment]
@pizzatiger what do you mean by this

File: code/modules/admin/verbs/randomverbs.dm
Comment (MrDoomBringer) [reply code/modules/admin/verbs/randomverbs.dm]: goddammit i fixed this at home but forgot to commit ree

**Comment by MrDoomBringer** [comment]
@optimumtact sir this is a wip please read the sign

File: code/modules/admin/verbs/randomverbs.dm
Comment (octareenroon91) [reply code/modules/admin/verbs/randomverbs.dm]: Reminder to commit

Comment (MrDoomBringer) [reply code/modules/admin/verbs/randomverbs.dm]: image

holy fuck it keeps happening

**Comment by pizzatiger** [comment]
Like they send some sort of beacon and a portal opens up, spilling out whatever goodies you wanted to send the crew before self destructing. I

**Comment by MrDoomBringer** [comment]
image
cult pod

**Comment by MrDoomBringer** [comment]
@obscolene ah fuck

**Comment by MrDoomBringer** [comment]
image
@obscolene i feel like it looks better upside down

**Comment by MrDoomBringer** [comment]
@obscolene ok its good now
image

**Comment by MrDoomBringer** [comment]
@optimumtact should be good fam
untested tho

**Comment by MrDoomBringer** [comment]
wait holup i fucked up the map merge

**Comment by MrDoomBringer** [comment]
image
updated menu, contains a few new features. Also supplypods come in at an angle now

**Comment by davethwave** [comment]
How does the random angle thing effect the gondola pod?

**Comment by MrDoomBringer** [comment]
@davethwave oh fuck hahaha

File: code/modules/cargo/expressconsole.dm
Comment (optimumtact) [reply code/modules/cargo/expressconsole.dm]: you still haven't done anything with it

File: code/modules/cargo/supplypod.dm
Comment (optimumtact) [inline code/modules/cargo/supplypod.dm]: the hell is this, it should be in a subtype

File: code/modules/cargo/supplypod.dm
Comment (optimumtact) [inline code/modules/cargo/supplypod.dm]: why async

File: code/modules/cargo/supplypod.dm
Comment (optimumtact) [inline code/modules/cargo/supplypod.dm]: less meme comments please

File: code/modules/cargo/expressconsole.dm
Comment (MrDoomBringer) [reply code/modules/cargo/expressconsole.dm]: >implying
fuck ok

File: code/modules/cargo/supplypod.dm
Comment (MrDoomBringer) [reply code/modules/cargo/supplypod.dm]: frequently this foolish fruit fails to fathom my fantastic functions

Comment (MrDoomBringer) [reply code/modules/cargo/supplypod.dm]: this means i have no fucking idea why i did that

File: code/modules/cargo/supplypod.dm
Comment (MrDoomBringer) [reply code/modules/cargo/supplypod.dm]: "less ... comments please" - optimumtact

File: code/modules/cargo/supplypod.dm
Comment (optimumtact) [reply code/modules/cargo/supplypod.dm]: then do not

Comment (MrDoomBringer) [reply code/modules/cargo/supplypod.dm]: @optimumtact wait i remember why i did
it was so i could call the setOpened proc on the atom/movable/holder (in our case the gondolapod mob) without having to code a setOpened proc into the atom class

Comment (MrDoomBringer) [reply code/modules/cargo/supplypod.dm]: i have a suspiciion that doing this was retarded what do u think

File: code/modules/cargo/gondolapod.dm
Comment (MrDoomBringer) [reply code/modules/cargo/gondolapod.dm]: @81Denton gondola pod is big and strong so it gets 2

**Comment by octareenroon91** [comment]
Travis fail due to `code/modules/cargo/supplypod.dm:98:warning: if: if statement has no effect`

File: code/modules/cargo/supplypod.dm
Comment (octareenroon91) [inline code/modules/cargo/supplypod.dm]: oh yeah, that's a seriously important tab that's missing.

**Comment by TheDreamweaver** [comment]
Add another walking pod thats just Cthulhu. Also, one thats just a rock

**Comment by davethwave** [comment]
I think it's better it leave it as it is when it comes to pods and fix up any remaining bugs to get this merged. If anyone wants to add more pods, well just wait until this gets merged and make another PR in my opinion.

**Comment by TheDreamweaver** [comment]
@davethwave good opinion :)

**Comment by MrDoomBringer** [comment]
@TheDreamweaver yeah oranges already mentioned to PR more sprites later. I feel like the cthulu one is a bit oddly specific, but i like the rock idea

**Comment by TheDreamweaver** [comment]
😍

**Comment by TheDreamweaver** [comment]
gimme gimme gimme

**Comment by octareenroon91** [comment]
CHADMINBUS

**Comment by TheDreamweaver** [comment]
Now you just got to do the build mode stuff.

**Comment by MrDoomBringer** [comment]
@TheDreamweaver oh yeah i forgot about that
done

**Comment by TheDreamweaver** [comment]
Nice. The time is nyeh

**Comment by Borisvanmemes** [comment]
The fact that this hasn't been merged yet pains me deeply.

File: code/modules/cargo/expressconsole.dm
Comment (optimumtact) [reply code/modules/cargo/expressconsole.dm]: Isn't this still wrong?

**Comment by davethwave** [comment]
Welp, time for some interesting stuff to happen once the servers update.

**Pull request merged**
